src/main.rs

```diff
@@ -16,28 +21,45 @@ struct VersionQuery {
 }
 
 struct AppData {
+    cache: Mutex<TimedCache<&'static str, CachedReleased>>,
     config: ApiConfig,
     fetcher: Fetcher,
 }
 
+#[derive(Clone)]
+enum CachedReleased {
+    Updater(HashMap<String, Asset>),
+    Game(Vec<GameRelease>),
+}
+
 #[get("/game_version")]
 async fn game_version(
     app_data: web::Data<AppData>,
     ver_query: web::Query<VersionQuery>,
 ) -> impl Responder {
-    let AppData { config, fetcher } = app_data.as_ref();
-
-    let Ok(updater_releases) = fetcher.get_updater_releases().await else {
+    let AppData {
+        cache,
+        config,
+        fetcher,
+    } = app_data.as_ref();
+    let mut cache = cache.lock().unwrap();
+
+    // TODO: remove .cloned
+    let Ok(CachedReleased::Updater(updater_releases)) = cache.try_get_or_set_with("updater_releases", || async {
+        fetcher.get_updater_releases().await.map(CachedReleased::Updater)
```

Comment:
wouldn't it be possible to do it with cached macros?

Reply:
it's possible, but the result will be the same problem (for the data) or the creation of 2 different cache structures.

Reply:
correction, I don't think it's possible to use macros after all because of the asyncrone